### PR TITLE
[master < T2387-CL] Add new GSS functions and directives

### PR DESCRIPTION
**Pull request by tonilastre**
### New GSS functions and directions

> **Important note:** Please do not merge this before we release the new version of the Lab which will contain all the updates to the GSS noted and documented in this PR.

It contains the documentation for the following updates:

* New GSS graph functions: `Edges`, `Nodes`
* New GSS string functions: `Slice`, `Replace`, `Trim`, `Split`
* New GSS type functions: `IsNumber`, `IsString`, `IsBoolean`, `IsNull`
* New GSS directives: `@GraphStyle`, `@GraphStyle.Map`

### Pull request type

- [x] New documentation page
- [x] Documentation improvements

### Checklist:

- [x] I checked all content with Grammarly
- [x] I performed a self-review of my code
- [x] I made corresponding changes to the rest of the documentation
- [x] My changes generate no new warnings or errors

### CODE REVIEW

File: lab/style-script/gss-functions.md [hunk 1]
```diff
@@ -1148,6 +1253,70 @@ Outputs:
 
 - `string`
 
+### `IsNumber(value)`
+
+Returns True if the input value is a number, otherwise False.
```
Comment (vpavicic) [inline lab/style-script/gss-functions.md:1258]: If it returns True or False, i would format it as code... if it returns true and false then they are normal words

File: lab/style-script/gss-graphstyle-directive.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
+id: gss-graphstyle-directive
+title: Graph Style Script @GraphStyle directive properties
+sidebar_label: GraphStyle directive properties
+---
```
Comment (vpavicic) [inline lab/style-script/gss-graphstyle-directive.md:5]: Can you give an intro?

File: lab/style-script/gss-graphstyle-directive.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## `@GraphStyle`
+
+Here is the list of all properties that can be defined in the `@GraphStyle` directive,
+along with their expected types.
+
+### `collision-radius: number`
+
+Sets the margin radius for each node from its centre. If node `size` is `10` and
+`collision-radius` is set to 20, it means there will be `10` spaces left around each
```
Comment (vpavicic) [inline lab/style-script/gss-graphstyle-directive.md:15]: Why is 10 formatted as code, and 20 is not?

File: lab/style-script/gss-graphstyle-map-directive.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+Here is the list of all properties that can be defined in the `@GraphStyle.Map`
+directive, along with their expected types.
+
+### 1. `tile-layer: string: "detailed" | "light" | "dark" | "basic" | "satellite"`
+
+Sets the map tile for the map background. The default map tile is `"light"`.
+
+Examples:
+- `tile-layer: "dark"` sets the map tile to be type "dark".
```
Comment (vpavicic) [inline lab/style-script/gss-graphstyle-map-directive.md:17]: code formatting for "dark"?

File: lab/style-script/gss-graphstyle-map-directive.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+---
+id: gss-graphstyle-map-directive
+title: Graph Style Script @GraphStyle.Map directive properties
+sidebar_label: GraphStyle.Map directive properties
+---
+
```
Comment (vpavicic) [inline lab/style-script/gss-graphstyle-map-directive.md:6]: This is a new page? can we have a little intro like the definition you stated in the reference-guide?

File: lab/style-script/reference-guide.md [hunk 6]
```diff
@@ -216,6 +220,127 @@ Take a look at the [GSS @EdgeStyle directive
...
+  ...
+  <property-name-n>: <value expression-n>
+}
+```
+
+Similar to `@NodeStyle` and `@EdgeStyle`, `@GraphStyle` has a built-in variable
+`graph` which can be used for directive filter or property assignment.
+
+An example below shows a general directive style definition and a directive where
+style properties will be only applied if there are more than 10 nodes in the graph.
```
Comment (vpavicic) [inline lab/style-script/reference-guide.md:243]:
```suggestion
style properties will only be applied if there are more than 10 nodes in the graph.
```

**Review by vpavicic — CHANGES_REQUESTED** [review]
Resolve comments please :)

File: lab/style-script/gss-graphstyle-directive.md (hunk 3 above)
Comment (tonilastre) [reply lab/style-script/gss-graphstyle-directive.md:15]: Because I am blind :D Will fix it :D

File: lab/style-script/gss-functions.md (hunk 1 above)
Comment (vpavicic) [reply lab/style-script/gss-functions.md:1258]: @tonilastre - you don't agree with this or? This is the only change missing...

Comment (tonilastre) [reply lab/style-script/gss-functions.md:1258]: Ups sorry, totally missed this one. I will change it to code formatted values right away.

**Review by vpavicic — APPROVED** [review]

**Comment by vpavicic** [comment]
@tonilastre - what is the ATA for this to be merged?

**Comment by tonilastre** [comment]
> @tonilastre - what is the ATA for this to be merged?

in two days, Wednesday.

**Pull request merged**